core-plugins/src/main/java/io/cdap/plugin/transform/ProjectionTransform.java

```diff
@@ -138,6 +145,20 @@ public void initialize(TransformContext context) throws Exception {
     collector.getOrThrowException();
   }
 
+  @Override
+  public void prepareRun(StageSubmitterContext context) {
+    init(context.getInputSchema(), context.getFailureCollector());
+
+    List<FieldOperation> output = TransformLineageRecorderUtils.generateOneToOnes(new ArrayList<>(fieldsToKeep),
```

Comment:
Like the transform plugin, initialize the list first in case the util returns an immutable list